validator/validator.go

```diff
@@ -12,27 +13,53 @@ import (
 	"github.com/filecoin-project/go-state-types/big"
 )
 
-// NewValidatorFromString parses a string containing of validator address and multiaddress separated by "@".
+// NewValidatorFromString creates a validator based on the string in `Addr:Weight@NetworkAddr` format.
+//
+// An examples of a validator string:
+//   - t1wpixt5mihkj75lfhrnaa6v56n27epvlgwparujy:10@/ip4/127.0.0.1/tcp/10000/p2p/12D3KooWJhKBXvytYgPCAaiRtiNLJNSFG5jreKDu2jiVpJetzvVJ
 //
-// Example of validator strings:
-// - t1wpixt5mihkj75lfhrnaa6v56n27epvlgwparujy@/ip4/127.0.0.1/tcp/10000/p2p/12D3KooWJhKBXvytYgPCAaiRtiNLJNSFG5jreKDu2jiVpJetzvVJ
 // FIXME: Consider using json serde for this to support multiple multiaddr for validators.
 func NewValidatorFromString(s string) (*Validator, error) {
+	var w int64
+
 	parts := strings.Split(s, "@")
 	if len(parts) != 2 {
-		return nil, fmt.Errorf("incorrect format of the string")
+		return nil, fmt.Errorf("failed to parse validators string")
+	}
+	idAndWeight := parts[0]
+	netAddr := parts[1]
+	parts = strings.Split(idAndWeight, ":")
+	if len(parts) > 2 {
+		return nil, fmt.Errorf("weight or ID are incorrect")
+	}
+
+	id := parts[0]
+	if len(parts) == 2 {
+		n, err := strconv.ParseInt(parts[1], 10, 64)
```

Comment:
The actual values that will be used for the weight in the actor won't generally fit into 64 bits, so the files won't be able to express memberships with a lot of collateral (more than a few FIL).
This is probably not an issue for functionality, since they will probably only be used for manual configuration. But it would still be better to support big integers here. If that's too tedious, let's at least put a comment about this limitation in the code and in whatever (current or future) documentation on the membership file.
The bigger danger is if someone tries to write a realistic number in the file an the and the parsing will overflow.

Reply:
Fixed